twtr/src/integration-test/groovy/twtr/AccountFunctionalTests/GetByIdOrHandleSpec.groovy

```diff
@@ -0,0 +1,134 @@
+
+    @Shared
+    def goodIds = [:]
+
+    RESTClient restClient
+
+    def setup() {
+        restClient = new RESTClient("http://localhost:8080")
+
+        addAccount('1')
+        addAccount('2')
+        addAccount('3')
+    }
+
+    def cleanup() {
+        goodIds.each { it -> restClient.delete(path: "/accounts/${it.value}") }
+        goodIds.clear()
+    }
+
+    def addAccount(String postfix) {
+        def response = restClient.post(path: '/accounts', contentType: 'application/json',
+                body: [handle      : goodHandle + postfix,
+                       emailAddress: goodEmailAccount + postfix + goodEmailDomain,
+                       password    : goodPassword,
+                       displayName : goodDisplayName + postfix])
+        goodIds << ["$postfix": response.data.id]
+    }
+
+    def 'get by valid account id: #description'() {
+        when:
+        def goodId = goodIds.find { it -> it.key == postfix }.value
+        def response = restClient.get(path: "/accounts/$goodId")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.id == goodId
+        response.data.handle == goodHandle + postfix
+        response.data.emailAddress == goodEmailAccount + postfix + goodEmailDomain
+        response.data.displayName == goodDisplayName + postfix
+        response.data.followerCount == 0
+        response.data.followingCount == 0
+        response.data.messageCount == 0
+
+        where:
+        description | postfix
+        'account 1' | '1'
+        'account 2' | '2'
+        'account 3' | '3'
+    }
+
+    def 'get by valid handle: #description'() {
+        when:
+        def goodId = goodIds.find { it -> it.key == postfix }.value
+        def response = restClient.get(path: "/account/handle/${goodHandle + postfix}")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.id == goodId
+        response.data.handle == goodHandle + postfix
+        response.data.emailAddress == goodEmailAccount + postfix + goodEmailDomain
+        response.data.displayName == goodDisplayName + postfix
+        response.data.followerCount == 0
+        response.data.followingCount == 0
+        response.data.messageCount == 0
+
+        where:
+        description | postfix
+        'account 1' | '1'
+        'account 2' | '2'
+        'account 3' | '3'
+    }
+
+    def 'get by invalid handle: #description'() {
+        when:
+        def response = restClient.get(path: "/account/handle/$handle")
+
+        then:
+        HttpResponseException problem = thrown(HttpResponseException)
+        problem.statusCode == 404
+
+        where:
+        description                    | handle
+        'same as a password'           | 'abc123ABC'
+        'same as handle prefix'        | 'scsu_huskies'
+        'one more than largest handle' | 'scsu_huskies4'
+        'starts the same as a handle'  | 'scsu_huskies10'
```

Comment:
oh, if i have an id that is scsu_huskies1, i don't want scsu_huskies10 to be returned.  So, essentially ruling out a situation where the controller might search using ==~ "$handle%".